test/integration/basic_test.go

```diff
@@ -0,0 +1,68 @@
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Check if opt-in/opt-out works
+func TestOptOutOptIn(t *testing.T) {
+	// Backup pull secret from openshift-config namespace.
+	// oc extract secret/pull-secret -n openshift-config --to=.
+	pullSecret, err := kubeClient.CoreV1().Secrets("openshift-config").Get("pull-secret", metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	// Edit the `.dockerconfigjson` file that was downloaded.
+	// Remove the `cloud.openshift.com` JSON entry.
+	newPullSecret := pullSecret.DeepCopy()
+	secretConfig := newPullSecret.Data[".dockerconfigjson"]
+
+	obj := map[string]interface{}{}
+	json.Unmarshal([]byte(secretConfig), &obj)
+	creds := obj["auths"].(map[string]interface{})
+	delete(creds, "cloud.openshift.com")
+
+	modifiedConfig, err := json.Marshal(obj)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	newPullSecret.Data[".dockerconfigjson"] = modifiedConfig
+
+	objOld := map[string]interface{}{}
+	json.Unmarshal([]byte(pullSecret.Data[".dockerconfigjson"]), &objOld)
+	credsOld := objOld["auths"].(map[string]interface{})
+	fmt.Println(credsOld)
```

Comment:
prob. `t.Log(credsOld)` might be better inside unit test

Reply:
oh, that part needs to be removed, it's not used